### DISCUSSION TITLE
Public function for `usdz` export

### DISCUSSION THREAD

**Discussion by annahassel** [google/model-viewer#4454] [Ideas]
Is it possible to add public function for `usdz` export? For iOS issue I need to save real `usdz` file, and it's will be great if this file will be same as generated `usdz` file from model-viewer. But model-viewer element has no ability to get generated file or `three.js` scene (that could be used for `USDZExporter` utility).

Are you interested in PR for this feature?

**Comment by elalish** [comment]
Hmm, can I understand the use case a little better first? Our USDZ export is not the best; is there a reason you wouldn't use Apple's tools to build a USDZ and then use that in our `ios-src`?

**Comment by annahassel** [comment]
It's all because `WKWebView does not support opening USDZs in QuickLook`, so I need to store `usdz` for iOS app. I want to convert `glb` to `usdz` in the user's browser. I am satisfied with the result from [`USDZExporter` utility](https://github.com/google/model-viewer/blob/master/packages/model-viewer/src/features/ar.ts#L18) and it solves my problem. At the moment I using `GLTFLoader` to get the `three.js` scene from user's `glb` and then pass it to `USDZExporter`, but that `usdz` looks different from the model-viewer. It will be great for me if these files are the same, but it's okay if it's not a model-viewer job.

**Reply by elalish** [reply]
Strange, that should give identical results. Perhaps check with three.js version you're using? What differences have you seen? You might look through our export function to see any tweaks we make to the scene you may want to copy.
